cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -258,34 +266,77 @@ Slice::Python::CodeVisitor::typeToTypeHintString(
+                {
+                    // For boolean sequences "python:array.array" is mapped to array.array('b'), whose type-hint
+                    // is array.array[int].
+                    os << " | " << getImportAlias(source, "array", "array") << "[int]";
+                }
+                else if (metadataDirective == "python:numpy.ndarray")
                 {
                     assert(elementType && elementType->kind() <= Builtin::KindDouble);
                     const string numpyAlias = getImportAlias(source, "numpy");
                     os << " | " << numpyAlias << ".typing.NDArray[" << numpyAlias << "."
                        << numpyBuiltinTable[elementType->kind()] << "]";
                 }
+                else if (sequenceMetadata && metadataDirective == "python:memoryview")
+                {
+                    auto [_, typeHint] = splitMemoryviewArguments(sequenceMetadata->arguments());
+                    if (typeHint)
+                    {
+                        auto [package, name] = splitFQN(*typeHint);
+                        auto memoryViewType = getImportAlias(source, package, name);
+                        os << " | " << memoryViewType;
+                    }
+                }
             }
-            else if (seq->hasMetadata("python:list") || seq->hasMetadata("python:seq:list"))
+            else if (metadataDirective == "python:list")
             {
                 os << "list[" << typeToTypeHintString(seq->type(), false, source, forMarshaling) << "]";
             }
-            else if (seq->hasMetadata("python:tuple") || seq->hasMetadata("python:seq:tuple"))
+            else if (metadataDirective == "python:tuple")
             {
                 os << "tuple[" << typeToTypeHintString(seq->type(), false, source, forMarshaling) << ", ...]";
             }
-            else if (seq->hasMetadata("python:numpy.ndarray"))
+            else if (metadataDirective == "python:numpy.ndarray")
             {
                 assert(elementType && elementType->kind() <= Builtin::KindDouble);
                 const string numpyAlias = getImportAlias(source, "numpy");
                 os << numpyAlias << ".typing.NDArray[" << numpyAlias << "." << numpyBuiltinTable[elementType->kind()]
                    << "]";
             }
-            else if (seq->hasMetadata("python:array.array"))
+            else if (metadataDirective == "python:array.array")
             {
                 assert(elementType && elementType->kind() <= Builtin::KindDouble);
-                const string arrayAlias = getImportAlias(source, "array");
-                os << arrayAlias << ".array[" << typeToTypeHintString(seq->type(), false, source, forMarshaling) << "]";
+                const string arrayAlias = getImportAlias(source, "array", "array");
+                // array.array does not support boolean values. The common convention is to use a
```

Comment:
I would update this comment to match the one above regarding array and bool seq.